### PR TITLE
Fixes #27994 - Clean up deprecated functions

### PR DESCRIPTION
**Pull request by tbrisker**

### CODE REVIEW

File: lib/foreman/renderer/scope/variables/base.rb [hunk 1]
```diff
@@ -75,7 +75,7 @@ def kickstart_attributes
             @arch      = architecture_name
             @osver     = major.try(:to_i)
             @mediapath = mediumpath(@medium_provider) if @medium_provider
-            @repos     = repos(host)
+            @repos     = @medium_provider&.additional_media
```
Comment (xprazak2) [inline lib/foreman/renderer/scope/variables/base.rb]: Is it possible that `@additional_media` are loaded for us already by `load_variables_base` on l.33?

Comment (tbrisker) [reply lib/foreman/renderer/scope/variables/base.rb]: good point. Actually, i think this variable should be removed along with the method, it doesn't look like it's being used in the default templates.

File: test/unit/foreman/renderer/scope/macros/snippet_rendering_test.rb [hunk 2]
```diff
@@ -40,7 +40,7 @@ class SnippetRenderingTest < ActiveSupport::TestCase
   test "should render a snippets with variables" do
```
Comment (xprazak2) [inline test/unit/foreman/renderer/scope/macros/snippet_rendering_test.rb]: nit: s/snippets/snippet/

File: test/unit/foreman/renderer/scope/macros/snippet_rendering_test.rb [hunk 3]
```diff
@@ -40,7 +40,7 @@ class SnippetRenderingTest < ActiveSupport::TestCase
...
     Foreman::Deprecation.expects(:deprecation_warning).once
```
Comment (xprazak2) [inline test/unit/foreman/renderer/scope/macros/snippet_rendering_test.rb]: This will probably need to be removed for the test to pass.

**Comment by xprazak2** [comment]
Almost there, test failures related.

File: test/unit/foreman/renderer/scope/macros/snippet_rendering_test.rb (hunk 3 above)
Comment (tbrisker) [reply test/unit/foreman/renderer/scope/macros/snippet_rendering_test.rb]: thanks!

**Comment by tbrisker** [comment]
[test foreman]

File: script/foreman-config [hunk 4]
```diff
@@ -1,48 +0,0 @@
-#!/usr/bin/env ruby
```
Comment (ekohl) [inline script/foreman-config:1]: Heads up: this file is mentioned in packaging

Comment (tbrisker) [reply script/foreman-config:1]: https://github.com/theforeman/foreman-packaging/pull/4176

File: app/assets/javascripts/application.js [hunk 5]
```diff
@@ -14,7 +14,6 @@
 //= require lookup_keys
 //= require editable/bootstrap-editable
 //= require editable/rails
-//= require deprecation
```
Comment (ohadlevy) [inline app/assets/javascripts/application.js:17]: why remove the entire file? does it make sense to leave it as a place to add new deprecated functions in the future?

Comment (tbrisker) [reply app/assets/javascripts/application.js:17]: Adding the file again if needed is easy, i would rather not keep an empty file that isn't used. I also don't mind if functions are deprecated in the existing files without moving them around for it.

Comment (ekohl) [reply app/assets/javascripts/application.js:17]: Could plugins be depending on this?

File: lib/tasks/puppet.rake [hunk 6]
```diff
@@ -23,140 +23,6 @@ namespace :puppet do
...
-          exit 1
-        end
-        puts "Import ends: #{Time.now.strftime('%Y-%m-%d %H:%M:%S %Z')}"
-        puts " ================================================================ "
-      end
-    end
-  end
 
   namespace :import do
     desc "
```
Comment (ekohl) [inline lib/tasks/puppet.rake:28]: I'm wondering if the part below is still relevant. Looks like the code doesn't handle class parameters so I doubt anyone's using this. Perhaps deprecate it in a separate PR?

File: lib/tasks/puppet.rake [hunk 7]
```diff
@@ -23,140 +23,6 @@ namespace :puppet do
       end
```
Comment (ekohl) [inline lib/tasks/puppet.rake:23]: For a separate PR, but I'd suggest deprecating the above in favor of `/etc/puppetlabs/puppet/node.rb --push-facts` which provides the same functionality and even has a `--push-facts-parallel` option.

File: app/assets/javascripts/application.js (hunk 5 above)
Comment (tbrisker) [reply app/assets/javascripts/application.js:17]: I don't think so, this is part of the core assets. If they are using the functions in this file they should stop as they are being removed.

File: script/foreman-config (hunk 4 above)
Comment (tbrisker) [reply script/foreman-config:1]: packaging pr was merged.

File: app/assets/javascripts/application.js (hunk 5 above)
Comment (ekohl) [reply app/assets/javascripts/application.js:17]: I've mistaken and thought deprecation included the functions to mark it as deprecated. However, that's `tfm.tools.deprecate` so ignore my comment.

File: lib/tasks/puppet.rake (hunk 6 above)
Comment (tbrisker) [reply lib/tasks/puppet.rake:28]: PRs welcome :wink:

**Comment by tbrisker** [comment]
Also added to the upgrade warnings: https://github.com/theforeman/theforeman.org/pull/1477

**Review by ekohl — APPROVED** [review]

**Comment by ekohl** [comment]
[test rpm]

**Comment by mmoll** [comment]
[test katello]

**Comment by mmoll** [comment]
[test foreman]

**Comment by mmoll** [comment]
merged, thanks @tbrisker!

**Pull request merged**